CyborgianStates/Commands/CouldEndorseCommand.cs

```diff
@@ -0,0 +1,179 @@
+﻿using CyborgianStates.CommandHandling;
+using CyborgianStates.Data.Models.Dump;
+using CyborgianStates.Interfaces;
+using CyborgianStates.MessageHandling;
+using CyborgianStates.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Quartz;
+using Serilog;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CyborgianStates.Commands
+{
+    public class CouldEndorseCommand : ICommand
+    {
+        private readonly AppSettings _config;
+        private readonly IResponseBuilder _responseBuilder;
+        private readonly IDumpDataService _dumpDataService;
+        private readonly ILogger _logger;
+        private CancellationToken token;
+        
+        public CouldEndorseCommand() : this(Program.ServiceProvider)
+        {
+        }
+
+        public CouldEndorseCommand(IServiceProvider serviceProvider)
+        {
+            _logger = Log.ForContext<CouldEndorseCommand>();
+            _config = serviceProvider.GetRequiredService<IOptions<AppSettings>>().Value;
+            _responseBuilder = serviceProvider.GetRequiredService<IResponseBuilder>();
+            _dumpDataService = serviceProvider.GetRequiredService<IDumpDataService>();
+        }
+
+        public async Task<CommandResponse> Execute(Message message)
+        {
+            if (message is null)
+            {
+                throw new ArgumentException(nameof(message));
+            }
+
+            try
+            {
+                _logger.Debug(message.Content);
+                var parameters = message.Content.Split(" ").Skip(1);
+
+                // Checks for parameter count and extracts the first parameter or throws an exception.
+                string nation = parameters.Count() == 1
+                    ? Helpers.ToID(string.Join(" ", parameters))
+                    : throw new Exception();
+                
+                await ProcessResultAsync(message, nation).ConfigureAwait(true);
+                
+                CommandResponse commandResponse = _responseBuilder.Build();
+                await message.Channel.ReplyToAsync(message, commandResponse).ConfigureAwait(false);
+                return commandResponse;
+            }
+            catch (InvalidOperationException e)
+            {
+                _logger.Error(e.ToString());
+                return await FailCommandAsync(message, "Specified nation is not a WA member.").ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                _logger.Error(e.ToString());
```

Comment:
Preferably you log a exception using a overload that takes the exception as a parameter.

Reply:
I stole this implementation from `RegionStatsCommand.cs` and `NationStatsCommand.cs`.  Do we need to modify the `ILogger` to allow overloading with an exception itself?

I am correcting it so that proper templating is used.